api/submission/remove_submission.go

```diff
@@ -0,0 +1,40 @@
+package submission
+
+import (
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/db"
+)
+
+type RemoveSubmissionRequest struct {
+    core.APIRequestAssignmentContext
+    core.MinRoleGrader
+
+    TargetUser core.TargetUserSelfOrGrader `json:"target-email"`
+    TargetSubmission string `json:"target-submission"`
+}
+
+type RemoveSubmissionResponse struct {
+    FoundUser bool `json:"found-user"`
+    FoundSubmission bool `json:"found-submission"`
+}
+
+func HandleRemoveSubmission(request *RemoveSubmissionRequest) (*RemoveSubmissionResponse, *core.APIError){
+    response := RemoveSubmissionResponse{};
+
+    if (!request.TargetUser.Found) {
+        return &response, nil;
+    }
+
+    response.FoundUser = true;
+
+    doesExist, err := db.RemoveSubmission(request.Assignment, request.TargetUser.Email, request.TargetSubmission);
+    if ( (err != nil) || (doesExist == false)){
+        response.FoundSubmission = false;
+        return nil, core.NewInternalError("-418", &request.APIRequestCourseUserContext, "Failed to remove the submission.").
```

Comment:
NEVER* group error and non-error conditions.
If you look at endpoints like peek and even your own response structure,
it is not an error to try and remove a submission that doesn't exist (like peeking an error that doesn't exist).

We want errors from our server to be as rare as possible, this means that we only create API errors when there is an actual error that we cannot recover from.
An unknown submission is not an error we can't recover from, we just tell them it doesn't exist.

**IMPORTANT**
I may be getting a bit too deep in my analysis here,
but I think it is important that you slow down a bit and really think about this paragraph.
Every piece of code should be intentional and make as much sense as possible.
Here you invoke an "InternalError" when a submission was not found.
But the submission ID is provided by the user, so the cause of that error cannot be internal, it's external (the user sent it to us).
Therefore, an "internal" error does not make sense.
I want to make sure you are thinking in ways like this when working with the codebase.

*99.9% of the time.